pcsx2/gui/Dialogs/GSDumpDialog.cpp

```diff
@@ -0,0 +1,778 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2020  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "App.h"
+#include "AppCommon.h"
+#include "MSWstuff.h"
+
+#include "Dialogs/ModalPopups.h"
+
+
+#include "Utilities/EmbeddedImage.h"
+#include "Resources/NoIcon.h"
+#include "GS.h"
+
+#include "PathDefs.h"
+#include "AppConfig.h"
+#include "Plugins.h"
+
+#include <wx/mstream.h>
+#include <wx/listctrl.h>
+#include <wx/filepicker.h>
+#include <wx/radiobut.h>
+#include <wx/button.h>
+#include <wx/treectrl.h>
+#include <wx/checkbox.h>
+#include <wx/dir.h>
+#include <wx/image.h>
+#include <wx/wfstream.h>
+#include <functional>
+
+using namespace pxSizerFlags;
+
+// --------------------------------------------------------------------------------------
+//  GSDumpDialog  Implementation
+// --------------------------------------------------------------------------------------
+
+Dialogs::GSDumpDialog::GSDumpDialog(wxWindow* parent)
+	: wxDialogWithHelpers(parent, _("GSDumpGov"), pxDialogFlags())
+	, m_dump_list(new wxListView(this, ID_DUMP_LIST, wxDefaultPosition, wxSize(400, 300), wxLC_NO_HEADER | wxLC_REPORT | wxLC_SINGLE_SEL))
+	, m_preview_image(new wxStaticBitmap(this, wxID_ANY, wxBitmap(EmbeddedImage<res_NoIcon>().Get()), wxDefaultPosition, wxSize(400,250)))
+	, m_debug_mode(new wxCheckBox(this, ID_DEBUG_MODE, _("Debug Mode")))
+	, m_renderer_overrides(new wxRadioBox())
+	, m_gif_list(new wxTreeCtrl(this, ID_SEL_PACKET, wxDefaultPosition, wxSize(400, 300), wxTR_HIDE_ROOT | wxTR_HAS_BUTTONS | wxTR_LINES_AT_ROOT))
+	, m_gif_packet(new wxTreeCtrl(this, wxID_ANY, wxDefaultPosition, wxSize(400, 300), wxTR_HIDE_ROOT | wxTR_HAS_BUTTONS | wxTR_LINES_AT_ROOT))
+	, m_start(new wxButton(this, ID_RUN_START, _("Go to Start"), wxDefaultPosition, wxSize(150,50)))
+	, m_step(new wxButton(this, ID_RUN_START, _("Step"), wxDefaultPosition, wxSize(150, 50)))
+	, m_selection(new wxButton(this, ID_RUN_START, _("Run to Selection"), wxDefaultPosition, wxSize(150, 50)))
+	, m_vsync(new wxButton(this, ID_RUN_START, _("Go to next VSync"), wxDefaultPosition, wxSize(150, 50)))
+	, m_thread(std::make_unique<GSThread>(this))
+{
+	//TODO: figure out how to fix sliders so the destructor doesn't segfault
+	wxBoxSizer& dump_info(*new wxBoxSizer(wxVERTICAL));
```

Comment:
This leaks memory every time the dialog is created? Also, I think your sizers are going out of scope when you declare them as references like this, creating a "use after free" scenario?

Reply:
The dialog is only created once, moreso wxwidgets destructor goes through everything listed in the *this object and deletes all of them, I ran into a bug related to that a few commits above :p

Reply:
And yeah, no, there is no UAF there; a unique reference to a pointer is created which is then freed on object deletion, nothing wrong with this code

Reply:
ohhh i left the todo there, might have confused you, let me remove it